tests/shared/src/test/scala/munit/DoubleAssertionsSuite.scala

```diff
@@ -0,0 +1,126 @@
+package munit
+
+class DoubleAssertionsSuite extends BaseSuite {
+  test("Assert Equals NaN Fails") {
+    try {
+      assertEquals(1.234, Double.NaN, 0.0)
+      fail("unexpected")
+    } catch {
+      case _: FailException =>
+    }
+  }
```

Comment:
You can use `.fail` to assert a test fails
```suggestion
  test("Assert Equals NaN Fails".fail) {
    assertEquals(1.234, Double.NaN, 0.0)
  }
```